base/ntuple.jl

```diff
@@ -16,22 +16,24 @@ julia> ntuple(i -> 2*i, 4)
 """
 @inline function ntuple(f::F, n::Integer) where F
     # marked inline since this benefits from constant propagation of `n`
-    t = n == 0  ? () :
-        n == 1  ? (f(1),) :
-        n == 2  ? (f(1), f(2)) :
-        n == 3  ? (f(1), f(2), f(3)) :
-        n == 4  ? (f(1), f(2), f(3), f(4)) :
-        n == 5  ? (f(1), f(2), f(3), f(4), f(5)) :
-        n == 6  ? (f(1), f(2), f(3), f(4), f(5), f(6)) :
-        n == 7  ? (f(1), f(2), f(3), f(4), f(5), f(6), f(7)) :
-        n == 8  ? (f(1), f(2), f(3), f(4), f(5), f(6), f(7), f(8)) :
-        n == 9  ? (f(1), f(2), f(3), f(4), f(5), f(6), f(7), f(8), f(9)) :
-        n == 10 ? (f(1), f(2), f(3), f(4), f(5), f(6), f(7), f(8), f(9), f(10)) :
-        _ntuple(f, n)
+    n_ = Int(n)::Int
```

Comment:
Could make the diff smaller by renaming the method argument from `n::Integer` to `m::Integer` and then doing `n = Int(m)::Int` instead of `n_ = Int(n)::Int`.